cpp/arcticdb/storage/library_manager.cpp

```diff
@@ -115,14 +119,61 @@ void LibraryManager::write_library_config(const py::object& lib_cfg, const Libra
     );
 }
 
+void LibraryManager::modify_library_option(const arcticdb::storage::LibraryPath &path,
+                                           std::variant<ModifiableLibraryOption, ModifiableEnterpriseLibraryOption> option,
+                                           std::variant<bool, int64_t> new_value) const {
+    // We don't apply a storage override to keep modification backwards compatible.
+    // Before v3.0.0 we didn't use storage overrides and applying one when modifying options would break readers older
+    // than v3.0.0.
+    arcticdb::proto::storage::LibraryConfig config = get_config_internal(path, std::nullopt);
+    auto mutable_write_options = config.mutable_lib_desc()->mutable_version()->mutable_write_options();
+
+    auto get_bool = [&option](std::variant<bool, int64_t> value){
+        util::check(std::holds_alternative<bool>(value), "{} only supports bool values but received {}. Not changing library option.", option, value);
```

Comment:
I think the `util::check` will bubble up to the user in Python as an `InternalException` - instead it should be one of the user-facing exception types

Reply:
Thanks, fixed.